src/PowerShellEditorServices/Services/Symbols/ReferenceTable.cs

```diff
@@ -55,62 +71,228 @@ internal void EnsureInitialized()
+        }
+
+        public override AstVisitAction VisitTypeExpression(TypeExpressionAst typeExpressionAst)
+        {
+            _references.AddReference(
+                SymbolType.Type,
+                typeExpressionAst.TypeName.Name,
+                typeExpressionAst.Extent,
+                typeExpressionAst.Extent);
+
+            return AstVisitAction.Continue;
+        }
+
+        public override AstVisitAction VisitTypeConstraint(TypeConstraintAst typeConstraintAst)
+        {
+            _references.AddReference(
+                SymbolType.Type,
+                typeConstraintAst.TypeName.Name,
+                typeConstraintAst.Extent,
+                typeConstraintAst.Extent);
+
+            return AstVisitAction.Continue;
+        }
+
+        public override AstVisitAction VisitFunctionMember(FunctionMemberAst functionMemberAst)
+        {
+            SymbolType symbolType = functionMemberAst.IsConstructor
+                ? SymbolType.Constructor
+                : SymbolType.Method;
+
+            IScriptExtent nameExtent = VisitorUtils.GetNameExtent(
+                functionMemberAst,
+                useQualifiedName: false,
+                includeReturnType: false);
+
+            _references.AddReference(
+                symbolType,
+                functionMemberAst.Name, // We bucket all the overloads.
+                nameExtent,
+                functionMemberAst.Extent,
+                isDeclaration: true);
+
+            return AstVisitAction.Continue;
+        }
+
+        public override AstVisitAction VisitPropertyMember(PropertyMemberAst propertyMemberAst)
+        {
+            SymbolType symbolType =
+                propertyMemberAst.Parent is TypeDefinitionAst typeAst && typeAst.IsEnum
+                    ? SymbolType.EnumMember : SymbolType.Property;
+
+            IScriptExtent nameExtent = VisitorUtils.GetNameExtent(propertyMemberAst, false, false);
+            _references.AddReference(
+                symbolType,
+                nameExtent.Text,
+                nameExtent,
+                propertyMemberAst.Extent,
+                isDeclaration: true);
+
+            return AstVisitAction.Continue;
+        }
+
+        public override AstVisitAction VisitMemberExpression(MemberExpressionAst memberExpressionAst)
+        {
+            string? memberName = memberExpressionAst.Member is StringConstantExpressionAst stringConstant ? stringConstant.Value : null;
+            if (string.IsNullOrEmpty(memberName))
+            {
+                return AstVisitAction.Continue;
+            }
+
+            _references.AddReference(
+                SymbolType.Property,
+                memberName,
+                memberExpressionAst.Member.Extent,
+                memberExpressionAst.Extent);
+
+            return AstVisitAction.Continue;
+        }
+
+        public override AstVisitAction VisitInvokeMemberExpression(InvokeMemberExpressionAst methodCallAst)
```

Comment:
Which scenario is this for? I don't think you'll be able to match it to the correct method to count it as a reference.

Reply:
So it turns out it works quite well! By bucketing all the overloads (of the same name) together, and returning a set of possible definitions, VS Code just does what it needs so you can do:

```powershell
class SuperClass : BaseClass {
    [string]MyClassMethod([string]$param1, $param2, [int]$param3) {
        $this.SomePropWithDefault = 'something happend'
        $param1
        return 'finished'
    }

    [string]
    MyClassMethod([MyEnum]$param1) {
        return 'hello world'
    }
    [string]MyClassMethod() {
        return 'hello world'
    }
}

$o = [SuperClass]::new()
$o.MyClassMethod()
```

and find definition on `MyClassMethod()` shows the three potential matches:
<img width="1681" alt="Screenshot 2023-01-20 at 5 48 12 PM" src="https://user-images.githubusercontent.com/2226434/213835617-cac4d752-3b3e-4e46-8966-a8c7c79985fb.png">

No, it's not perfect, but it's a _whole lot better_ than nothing!

Reply:
To make it complete, should `::new()` return the constructors? Because that's a reserved method when static (`InvokeMemberExpressionAst.Static == true`), right?

Reply:
That's a good idea!